src/main/java/com/purbon/kafka/topology/serdes/SystemPropertySubstitutor.java

```diff
@@ -0,0 +1,61 @@
+package com.purbon.kafka.topology.serdes;
+
+import com.purbon.kafka.topology.exceptions.TopologyParsingException;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.lookup.StrSubstitutor;
+import org.apache.logging.log4j.status.StatusLogger;
+
+/**
+ * A simple substitutor which substitutes system properties in the topology file. If a property is
+ * not found, it throws a {@link com.purbon.kafka.topology.exceptions.TopologyParsingException}.
+ */
+public class SystemPropertySubstitutor {
+  private final Map<String, String> env;
+  private final StrSubstitutor strSubstitutor;
+
+  public SystemPropertySubstitutor() {
+    env =
+        System.getProperties().entrySet().stream()
+            .collect(Collectors.toMap(e -> (String) e.getKey(), e -> (String) e.getValue()));
+
+    strSubstitutor =
+        new StrSubstitutor(env, "${", "}") {
+          @Override
+          protected String resolveVariable(
+              final LogEvent event,
+              final String variableName,
+              final StringBuilder buf,
+              final int startPos,
+              final int endPos) {
+            String result = super.resolveVariable(event, variableName, buf, startPos, endPos);
+            if (result == null) {
+              throw new TopologyParsingException("Cannot resolve variable: " + variableName);
+            }
+            return result;
+          }
+
+          @Override
+          public String replace(final LogEvent event, final String source) {
```

Comment:
Although I like the idea of supporting environment variables, I'm not sure I like using log4j for the implementation.

Reply:
Jsut for fun because the project is frozen, let's use Apache Commons Text StringSubstitutor instead. ;)

Reply:
It was a tad unfrozen yesterday. Let's hope it continues for a while. :-)